api-lib/gql/queries.ts

```diff
@@ -544,6 +544,10 @@ export async function getEpoch(
           token_gifts: [
             {},
             {
+              recipient: {
+                id: true,
+                deleted_at: true,
+              },
```

Comment:
is this needed, or can we just filter the data using the `where` on the backend graphql side and not do any FE side checking of the `deleted_at` field?

Reply:
Good question, to be fair I'm not sure. I wonder if the requirements have changed for the CSV, as we currently have the following
```typescript
{
  where: {
    _or: [
      { deleted_at: { _is_null: true } },
      { deleted_at: { _gt: epochEndDate } },
    ],
  },
}
```
So I'm assuming at some point we wanted the deleted users only if they were deleted after the epoch ended? In either case, I do agree with you that this filtering would be better done in the backend. Perhaps @zashton could shed some light here

Reply:
if the user is deleted after the epoch ends , the GIVES he received are still valid and not refunded so it should be represented somehow in the frontend e.g in the map etc so the accounting is correct.

Reply:
After reading through the issue i'm not sure this is a bug though.
Deleting a user only after the epoch ends can also mean the admin wants the user to be included in the distribution.
My understanding is that deleting a user should not affect any of the epochs that he has participated in the past, otherwise the allocation math would not add up.

Reply:
@zashton could the bug actually be in the distribution table rather than the csv? In the FE the GIVE that the deleted user got are not taken into account when calculating the percentages of the other users

Reply:
> Deleting a user only after the epoch ends can also mean the admin wants the user to be included in the distribution.
> My understanding is that deleting a user should not affect any of the epochs that he has participated in the past, otherwise the allocation math would not add up.

Hey @karelianpie - sorry for the confusion on this. I think we want the CSV export to match and have the exact same functionality as the Distribution page. Can you jsut wait a day or two for us to confirm and clarify the behavior this should have.

Reply:
But I think what you've done IS what we want, i.e. that the CSV excludes the deleted users GIVE and recalculates just like the distribution page.

Reply:
@karelianpie @zashton ok, so your original implementation has the correct and desired functionality for the CSV, which is to exclude deleted users and NOT show them as a line item in the CSV and recalculate the total give excluding their amount received. This follows the functionality of the Distributions page. So the only changes to this PR should be cleaning up the code, but the screenshots of the functionality are correct as done.

Reply:
Thank you @crabsinger for the clarification! I've refactored it so that it filters the deleted users directly in the `where` clause rather than doing it in the FE, please have a look https://github.com/coordinape/coordinape/pull/1367/commits/b3139fe985cd742069f421ebdadbeb41eaf654b4